src/opossum_lib/helper_methods.py

```diff
@@ -25,3 +27,71 @@ def _get_node_without_incoming_edge(graph: DiGraph) -> Optional[str]:
 
 def _node_represents_a_spdx_element(graph: DiGraph, node: str) -> bool:
     return "element" in graph.nodes[node]
+
+
+def _weakly_connected_component_sub_graphs(graph: DiGraph) -> List[DiGraph]:
+    connected_sub_graphs = []
+    for connected_set in weakly_connected_components(
+        graph
+    ):  # returns only a set of nodes without edges
+        connected_sub_graphs.append(graph.subgraph(connected_set).copy())
+
+    return connected_sub_graphs
+
+
+def _create_file_path_from_graph_path(graph: DiGraph, path: List[str]) -> str:
+    """
+    some file names with relative paths start with "./", to avoid paths like "/./"
+    we need to remove these prefixes
+    """
+    if list(graph.successors(path[-1])):
+        return (
+            "/"
+            + "/".join([graph.nodes[node]["label"].replace("./", "") for node in path])
+            + "/"
+        )
+    else:
+        return "/" + "/".join(
+            [graph.nodes[node]["label"].replace("./", "") for node in path]
+        )
+
+
+def _replace_node_ids_with_labels(
+    path: List[str], connected_subgraph: DiGraph
+) -> List[str]:
+    resulting_path = []
+    path_with_label = [
+        connected_subgraph.nodes[node]["label"].replace("./", "") for node in path
+    ]
+    for element_or_path in path_with_label:
+        resulting_path.extend(
+            [element for element in element_or_path.split("/") if element]
+        )
```

Comment:
I'm confused: Shouldn't the input parameter `path: List[str]` already be split into the path components? Why do we need to split on `/` a second time? Maybe that's worth a comment.

Reply:
The problem here is that some names from `SPDX Files` include a file path themselves. The current implementation uses the names from all spdx elements to build the file path tree so that instead of the SPDX identifiers in the previous implementation we see the actual names on the left side in OpossumUI. If we don't split these names again we would get folders with file paths. This is also the reason why I replaced the `tree_to_dict` method.